### PR TITLE
feat(application-system): Better DX when adding payments to an application.

### PR DESCRIPTION
**Pull request by obmagnusson**
## What

- Shared payment form for payment states.
- Create charge Shared template api 
- VerifyPayment Shared template api
- Documentation updated

## Why

Less code duplication for payments. Less implementation time to add payments for applications 

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by egill15 — APPROVED** [review]
LGTM

**Review by kksteini — APPROVED** [review]
CODEOWNER files LGTM

File: libs/application/types/src/lib/template-api/shared-api/shared-api-definitions/payment-catalog.ts [hunk 1]
```diff
@@ -6,5 +6,8 @@ export interface PaymentCatalogParameters {
 
 export const PaymentCatalogApi = defineTemplateApi<PaymentCatalogParameters>({
   action: 'paymentCatalog',
-  namespace: 'PaymentCatalog',
+  namespace: 'Payment',
+  params: {
+    organizationId: '6509142520',
```
Comment (saevarma) [inline libs/application/types/src/lib/template-api/shared-api/shared-api-definitions/payment-catalog.ts:11]: Is this default `organizationId` value or is the application template always required to provide the id when defining their paymentapi? Just wondering if we could accidentally have Syslumadur on some payments that should belong to another organisation?

**Review by saevarma — APPROVED** [review]
Approved, just one question about the `PaymentCatalogApi` `organizationId`.

File: libs/application/types/src/lib/template-api/shared-api/shared-api-definitions/payment-catalog.ts (hunk 1 above)
Comment (obmagnusson) [reply libs/application/types/src/lib/template-api/shared-api/shared-api-definitions/payment-catalog.ts:11]: yeah good point. Ill remove the this id as the default and just throw an error if it is missing.

**Pull request merged**